packages/@aws-cdk/aws-apigateway/lib/rate-limited-api-key.ts

```diff
@@ -0,0 +1,132 @@
+import { Construct, IResource as IResourceBase, Resource } from '@aws-cdk/core';
+import { ApiKey } from './api-key';
+import { ResourceOptions } from "./resource";
+import { RestApi } from './restapi';
+import { QuotaSettings, ThrottleSettings, UsagePlan, UsagePlanPerApiStage } from './usage-plan';
+
+/**
+ * API keys are alphanumeric string values that you distribute to
+ * app developer customers to grant access to your API
+ */
+export interface IRateLimitedApiKey extends IResourceBase {
+  /**
+   * The API key ID.
+   * @attribute
+   */
+  readonly keyId: string;
+}
+
+/**
+ * Rate limiting settings for the api key.
+ */
+export interface RateLimitingSettings {
```

Comment:
Let's drop this interface and add all of these properties directly into the `RateLimitedApiKeyProps` interface.

Reply:
Done!